adage/analyze/api.py

```diff
@@ -234,6 +237,20 @@ def get_annotations(request=None, **kwargs):
             'attachment; filename="sample_annotations.tsv"')
         return response
 
+    def apply_filters(self, request, applicable_filters):
+        """
+        Implementation of "experiment" filter, which allows the API to
+        get all samples that are related to a given experiment (whose
+        primary key is "accession").
+        """
+        object_list = super(SampleResource, self).apply_filters(
+            request, applicable_filters)
+        experiment = request.GET.get('experiment', None)
+        if experiment:
+            samples = Experiment.objects.get(pk=experiment).sample_set.all()
```

Comment:
What happens if the experiment key passed in turns out to be invalid? Do we need error handling in here, or does TastyPie do something smart for us?